geoh5py/ui_json/forms.py

```diff
@@ -0,0 +1,499 @@
+                try:
+                    setattr(self, member, members.pop(member))
+                except BaseValidationError as err:
+                    error_list.append(err)
+
+        if error_list:
+            if len(error_list) == 1:
+                raise error_list.pop()
+            raise AggregateValidationError(self.name, error_list)
+
+        self._extra_members.update(members)
+
+    def validate(self):
+        """Validates data against the pool of enforcers."""
+        enforcers = EnforcerPool.from_validations(self.name, self.validations)
+        enforcers.validate(self.form())
+
+    @property
+    def valid_members(self) -> list[str]:
+        """Recognized form member names."""
+        exclusions = ["_extra_members", "_active_members"]
+        private_attrs = [k for k in self.__dict__ if k.startswith("_")]
+        return [k[1:] for k in private_attrs if k not in exclusions]
+
+    @property
+    def active(self) -> list[str]:
+        """
+        Returns names of active form members.
+
+        :return: List of active form members.  These will include any members
+            that were:
+                1. Provided during construction.
+                2. Updated through the 'register' method.
+                3. Updated through member setters.
+        """
+        active = ["value"] + self._active_members + list(self._extra_members)
+        return list(dict.fromkeys(active))  # Unique while preserving order
+
+    @classmethod
+    def is_form(cls, form: dict[str, Any]) -> bool:
+        """Returns True if form contains any identifier members."""
+        id_members = cls.identifier_members
+        form_members = MEMBER_KEYS.map(form)
+        return any(k in form_members for k in id_members)
+
+    @property
+    def value(self):
+        return self._value.value
+
+    def _set_value_parameter(self, value) -> Parameter:
+        """Handles value argument as either a Parameter or a value."""
+
+        if isinstance(value, Parameter):
+            value.name = "value"
+            out = value
+        else:
+            out = Parameter("value", value)
+
+        return out
+
+    def _allow_values_access(self):
+        """Valid members public attr accesses underlying parameter value."""
+        for member in self.valid_members:
+            if member not in dir(self):
+                setattr(self.__class__, member, ValueAccess(f"_{member}"))
+
+    def __str__(self):
+        return f"<{type(self).__name__}> : '{self.name}' -> {self.value}"
+
+    def __contains__(self, item):
+        return MEMBER_KEYS.map_key(item) in self.active
+
+
+class StringFormParameter(FormParameter):
+    """String parameter type."""
+
+    def __init__(self, name, value=None, **kwargs):
+        value = StringParameter("value", value=value)
+        super().__init__(name, value=value, **kwargs)
+
+
+class BoolFormParameter(FormParameter):
+    """Boolean parameter type."""
+
+    def __init__(self, name, value=None, **kwargs):

```

Comment:
Should this not be a bool by default? I would expect the default state to be False?

Reply:
done.